tests/components/updater/test_init.py

```diff
@@ -62,21 +62,44 @@ def test_new_version_shows_entity_startup(hass, mock_get_uuid, mock_get_newest_v
     assert res, "Updater failed to set up"
 
     yield from hass.async_block_till_done()
-
     assert hass.states.is_state("binary_sensor.updater", "unavailable")
     assert "newest_version" not in hass.states.get("binary_sensor.updater").attributes
     assert "release_notes" not in hass.states.get("binary_sensor.updater").attributes
 
+
+@asyncio.coroutine
+def test_rename_entity(hass, mock_get_uuid, mock_get_newest_version):
+    """Test if renaming the binary sensor works correctly."""
+    mock_get_uuid.return_value = MOCK_HUUID
+    mock_get_newest_version.return_value = mock_coro((NEW_VERSION, RELEASE_NOTES))
+
+    now = dt_util.utcnow()
+    later = now + timedelta(hours=1)
+    mock_utcnow.return_value = now
+
+    res = yield from async_setup_component(hass, updater.DOMAIN, {updater.DOMAIN: {}})
+    assert res, "Updater failed to set up"
+
+    yield from hass.async_block_till_done()
+    assert hass.states.is_state("binary_sensor.updater", "unavailable")
+    assert hass.states.is_state("binary_sensor.new_entity_id", "unavailable") is False
```

Comment:
This state shouldn't exist at all. Get the state and assert that it is `None`.

Reply:
Done